src/main/scala/realworld/api/Articles.scala

```diff
@@ -0,0 +1,44 @@
+package realworld.api
+
+import realworld.ArticlesRepo
+import zio.ZIO
+import zio.http._
+import zio.http.model.{Method, Status}
+import zio.json._
+
+object Articles {
+
+  final val Routes: Http[ArticlesRepo, Throwable, Request, Response] =
+    Http.collectZIO[Request] {
+      case req @ Method.GET -> BasePath / "articles" =>
+        for {
+          repo     <- ZIO.service[ArticlesRepo]
```

Comment:
You can add an accessor method to `ArticlesRepo` if you plan to use it this way, e.g.:

```scala
object ArticlesRepo {
  def fetchAll(limit: Int, offset: Int): RIO[ArticlesRepo, List[Article]] =
    ZIO.serviceWithZIO(_.fetchAll(limit, offset))
}
```

With that in place, you can simplify the call site to:

```scala
ArticlesRepo.fetchAll(req.limit, req.offset)
```